pkg/beacon/beacon.go

```diff
@@ -23,21 +25,25 @@ var logger = log.Logger("keep-beacon")
 // internal random beacon implementation. Returns an error if this failed,
 // otherwise enters a blocked loop.
 func Initialize(
+	ctx context.Context,
 	operatorPublicKey *operator.PublicKey,
-	beaconChain beaconchain.Interface,
+	// TODO: Get rid of legacy Ethereum chain
+	legacyBeaconChain beaconchain.Interface,
+	// TODO: Accept a generic interface, not just sortition.Chain
+	beaconChain sortition.Chain,
```

Comment:
PR #3076 uses `beaconchain.Interface` as the type of the new `beaconChain`. Do we agree that `sortition.Chain` should be embedded into `beaconchain.Interface`?